x-pack/plugins/ml/public/application/routing/routes/trained_models/models_list.tsx

```diff
@@ -48,7 +49,20 @@ export const modelsListRouteFactory = (
 });
 
 const PageWrapper: FC = () => {
-  const { context } = useRouteResolver('full', ['canGetTrainedModels'], basicResolvers());
+  const { initSavedObjects } = useSavedObjectsApiService();
+
+  const initSavedObjectsWrapper = useCallback(async () => {
+    try {
+      await initSavedObjects();
+    } catch (error) {
+      // ignore error as user may not have permission to sync
+    }
+  }, [initSavedObjects]);
+
+  const { context } = useRouteResolver('full', ['canGetTrainedModels'], {
+    ...basicResolvers(),
+    initSavedObjectsWrapper,
+  });
```

Comment:
nit: `useRouteResolver` takes care of preserving refs for provided callbacks:

```suggestion
  const { initSavedObjects } = useSavedObjectsApiService();

  const { context } = useRouteResolver('full', ['canGetTrainedModels'], {
    ...basicResolvers(),
    initSavedObjects: () => initSavedObjects().catch(() => {})
  });
```

Reply:
This change is neater. I'm currently working on a related change which will do something similar on all ML pages. So for the sake of not having to retest this PR I'm going to merge as is and apply this refactor to the follow up PR